common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/BlockStoreClient.java

```diff
@@ -161,6 +172,22 @@ public void getHostLocalDirs(
       String[] execIds,
       CompletableFuture<Map<String, String[]>> hostLocalDirsCompletable) {
     checkInit();
+    int maxRetries = transportConf.maxIORetries();
+    int retryWaitTime = transportConf.ioRetryWaitTimeMs();
+    boolean enableSaslRetries = transportConf.enableSaslRetries();
+    retry(0, 0, maxRetries, enableSaslRetries, retryWaitTime, () -> {
+      CompletableFuture<Map<String, String[]>> tempHostLocalDirsCompletable =
+              new CompletableFuture<>();
+      getHostLocalDirsInternal(host, port, execIds, tempHostLocalDirsCompletable);
+      return tempHostLocalDirsCompletable;
+    }, hostLocalDirsCompletable);
+  }
+
+  private void getHostLocalDirsInternal(
+      String host,
+      int port,
+      String[] execIds,
+      CompletableFuture<Map<String, String[]>> hostLocalDirsCompletable) {
```

Comment:
While it does minimize the diff, it can make it slightly confusing as things evolve - `hostLocalDirsCompletable` which is passed into `retry`, and keeps propagating to subsequent executions of `retry` is different from the `hostLocalDirsCompletable` passed in as input here.

I would suggest refactoring this method to return the future - and not have it passed in as parameter.


```suggestion
  private CompletableFuture<Map<String, String[]>> getHostLocalDirsInternal(
      String host,
      int port,
      String[] execIds) {
      
     CompletableFuture<Map<String, String[]>> result = new CompletableFuture<>();
```

Reply:
Thanks. Done.